lib/rubocop/cop/rails/find_by.rb

```diff
@@ -31,7 +31,12 @@ class FindBy < Base
         MSG = 'Use `find_by` instead of `where.%<method>s`.'
         RESTRICT_ON_SEND = %i[first take].freeze
 
+        def_node_matcher :preceding_where?, <<~PATTERN
```

Comment:
Initially the matcher was called `where_first` but I thought it may be confused with the `ignore_where_first?` method.
Would love to hear better name suggestions ❤️

Reply:
I overlooked this case 😅 I think it's better to replace it with logic equivalent to `SCOPE_METHODS` instead of `def_node_matcher`.
https://github.com/rubocop/rubocop-rails/blob/v2.11.0/lib/rubocop/cop/rails/find_each.rb#L33

Besides `where`, there are `all`, `eager_load`, `includes`, and so on.

Reply:
Ah, interesting, we can use the `SCOPE_METHODS` approach, but If I'm not mistaken for our case it would be simply `SCOPE_METHODS = %i[where]` and that's it. Because `where().take` and `where().first` are the only two cases we want to check in the scope of this cop. Like we can't suggest `find_by` in cases like `User.all.take(5)` so we need just to skip those

Let me know if I'm missing something. Thanks!

Reply:
I've rebased on master and refactored it to avoid defining the `def_node_matcher` and explicitly check for `receiver&.method?(:where)`
I also removed the `receiver.block_type?` check because it should be covered by the `unless receiver&.method?(:where)` check

Reply:
Ah, you're right! Thank you!